subsys/net/lib/wifi_provision/wifi_provision.c

```diff
@@ -0,0 +1,995 @@
+
+		/* Signal strength */
+		scan.results[scan.results_count].rssi = entry->rssi;
+
+		scan.results_count++;
+	}
+}
+
+static void dhcp_server_start(void)
+{
+	int ret;
+	uint32_t address_int;
+	struct in_addr address;
+	struct in_addr netmask;
+	struct net_if *iface = net_if_get_first_wifi();
+
+	/* Set base address for DHCPv4 server based on statically assigned IPv4 address. */
+	if (inet_pton(AF_INET, CONFIG_WIFI_PROVISION_IPV4_ADDRESS, &address) != 1) {
+		LOG_ERR("Failed to convert IPv4 address");
+		notify_app(WIFI_PROVISION_EVT_FATAL_ERROR);
+		return;
+	}
+
+	/* Manually set IPv4 address */
+	if (net_if_ipv4_addr_add(iface, &address, NET_ADDR_OVERRIDABLE, 0) == NULL) {
+		LOG_ERR("Failed to add IPv4 address to interface");
+		notify_app(WIFI_PROVISION_EVT_FATAL_ERROR);
+		return;
+	}
+
+	LOG_DBG("IPv4 address added to interface");
+
+	/* Manually set IPv4 netmask. */
+	if (inet_pton(AF_INET, CONFIG_WIFI_PROVISION_IPV4_NETMASK, &netmask) != 1) {
+		LOG_ERR("Failed to convert netmask");
+		notify_app(WIFI_PROVISION_EVT_FATAL_ERROR);
+		return;
+	}
+
+	net_if_ipv4_set_netmask(iface, &netmask);
+
+	LOG_DBG("IPv4 netmask set");
+
+	/* Increment base address by 1 to avoid collision with the base address. */
+	address_int = ntohl(address.s_addr);
+	address_int += 1;
+	address.s_addr = htonl(address_int);
+
+	ret = net_dhcpv4_server_start(iface, &address);
+	if (ret) {
+		LOG_ERR("Failed to start DHCPv4 server, error: %d", ret);
+		notify_app(WIFI_PROVISION_EVT_FATAL_ERROR);
+		return;
+	}
+
+	LOG_DBG("DHCPv4 server started");
+}
+
+static void net_mgmt_wifi_event_handler(struct net_mgmt_event_callback *cb, uint32_t mgmt_event,
+					struct net_if *iface)
+{
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_AP_ENABLE_RESULT:
+		LOG_DBG("NET_EVENT_WIFI_AP_ENABLE_RESULT");
+
+		new_event(EVENT_AP_ENABLE);
+		break;
+	case NET_EVENT_WIFI_AP_DISABLE_RESULT:
+		LOG_DBG("NET_EVENT_WIFI_AP_DISABLE_RESULT");
+
+		new_event(EVENT_AP_DISABLE);
+		break;
+	case NET_EVENT_WIFI_AP_STA_CONNECTED:
+		LOG_DBG("NET_EVENT_WIFI_AP_STA_CONNECTED");
+
+		notify_app(WIFI_PROVISION_EVT_CLIENT_CONNECTED);
+		break;
+	case NET_EVENT_WIFI_AP_STA_DISCONNECTED:
+		LOG_DBG("NET_EVENT_WIFI_AP_STA_DISCONNECTED");
+
+		notify_app(WIFI_PROVISION_EVT_CLIENT_DISCONNECTED);
+		break;
+	case NET_EVENT_WIFI_SCAN_RESULT:
+		handle_wifi_scan_result(cb);
+		break;
+	case NET_EVENT_WIFI_SCAN_DONE:
+		LOG_DBG("NET_EVENT_WIFI_SCAN_DONE");
+
+		/* Scan results cached, waiting for client to request available networks and
+		 * provide Wi-Fi credentials.
+		 */
+		new_event(EVENT_SCAN_DONE);
+		break;
+	default:
+		break;
+	}
+}
+
+static int wifi_scan(void)
+{
+	int ret;
+	struct net_if *iface = net_if_get_first_wifi();
+	struct wifi_scan_params params = { 0 };
+
+	LOG_DBG("Scanning for Wi-Fi networks...");
+
+	/* Clear scan result buffer */
+	memset(scan_result_buffer, 0, sizeof(scan_result_buffer));
+
+	ret = net_mgmt(NET_REQUEST_WIFI_SCAN, iface, &params, sizeof(params));
```

Comment:
Side question - what's the intent of scanning in the provisioning library?

Reply:
Apparently on iOS application doesnt have access to APIs that lists available wifi networks or something like that.
@roshanrajaratnam can you elaborate?

Also it might be different what the device sees and the client that wants to provisioned it, although I suspect this is quite the rare case.

Reply:
This is more or less to identify what Wi-Fi networks are visible to the DK and not necessarily what's supported by the Android/iOS aps.

Reply:
Should this be a part of the library though? It doesn't seem to be essential to the provisioning process, and I imagine getting console logs isn't really useful for actual products.